buildchain/buildchain/docker_command.py

```diff
@@ -0,0 +1,300 @@
+            return TaskError(err)
+        return True
+
+
+# The call method is not counted as a public method
+# pylint: disable=too-few-public-methods
+class DockerTag:
+    """A class to expose the `docker tag` command through the API client."""
+
+    def __init__(self, repository: str, full_name: str, version: str):
+        """Initialize a `docker tag` callable object.
+
+        Arguments:
+            repository: the repository to which the tag should be pushed
+            full_name:  the fully qualified image name
+            version:    the version to tag the image with
+        """
+        self.repository = repository
+        self.full_name = full_name
+        self.version = version
+
+    def __call__(self) -> Optional[TaskError]:
+        try:
+            to_tag = DOCKER_CLIENT.images.get(self.full_name)
+            to_tag.tag(self.repository, tag=self.version)
+        except (docker.errors.BuildError, docker.errors.APIError) as err:
+            return doit.exceptions.TaskError(err)
+        return True
+
+
+# The call method is not counted as a public method
+# pylint: disable=too-few-public-methods
+class DockerPull:
+    """A class to expose the `docker pull` command through the API client."""
+
+    def __init__(self, repository: str, digest: str):
+        """Initialize a `docker pull` callable object.
+
+        Arguments:
+            repository: the repository to pull from
+            digest:     the digest to pull from the repository
+        """
+        self.repository = repository
+        self.digest = digest
+
+    def __call__(self) -> Optional[TaskError]:
+        try:
+            DOCKER_CLIENT.images.pull(self.repository, tag=self.digest)
+        except (docker.errors.BuildError, docker.errors.APIError) as err:
+            return doit.exceptions.TaskError(err)
+        return True
+
+
+# The call method is not counted as a public method
+# pylint: disable=too-few-public-methods
+class DockerSave:
+    """A class to expose the `docker save` command through the API client."""
+
+    def __init__(self, tag: str, save_path: Path):
+        """Initialize a `docker save` callable object.
+
+        Arguments:
+            tag:        the image's repository and tag
+            save_path:  the resulting image save path
+        """
+        self.tag = tag
+        self.save_path = save_path
+
+    def __call__(self) -> Optional[TaskError]:
+        try:
+            to_save = DOCKER_CLIENT.images.get(self.tag)
+            image_stream = to_save.save()
+            with self.save_path.open('wb') as image_file:
+                for chunk in image_stream:
+                    image_file.write(chunk)
+        except (docker.errors.APIError, IOError) as err:
+            return TaskError(err)
+        return True
+
+
+class DockerRun:
+    """A class to expose the `docker run` command through the API client."""
+
+    RPMLINTRC_MOUNT : Mount = Mount(
```

Comment:
You don't need mounts for any other commands.

Reply:
From f2f talk: if we import the module instead of specific classes, having mounts and mount helpers separate from the task class does indeed make a lot of sense. Will fix.